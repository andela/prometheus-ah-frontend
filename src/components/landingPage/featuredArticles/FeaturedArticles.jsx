import React from 'react';
import PropTypes from 'prop-types';
import FeaturedArticle from './FeaturedArticle';

const FeaturedArticles = (props) => {
  const { posts } = props;
  return (
    <div className="col-12 col-md-7 col-lg-7">
      <h2 className="section-title">FEATURED POSTS</h2>
      {
        posts.map((post, index) => (
          <FeaturedArticle
            key={post.id}
            active={index}
            image={post.image}
            title={post.title}
            date={post.createdAt}
            category={post.Tags[0]}
            description={post.description}
            readTime={post.readingTime}
            author={`${post.User.firstname} ${post.User.lastname}`}
          />
        ))
      }
    </div>
  );
};

FeaturedArticles.propTypes = {
  posts: PropTypes.instanceOf(Array)
};

export default FeaturedArticles;
