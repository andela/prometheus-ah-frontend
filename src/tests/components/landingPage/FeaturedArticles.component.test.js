import React from 'react';
import { shallow } from 'enzyme';
import FeaturedArticles from '../../../components/landingPage/featuredArticles/FeaturedArticles';
import mockData from '../../__mocks__/mockData';

let wrapper;

describe('LandingPage component', () => {
  beforeEach(() => {
    const { posts } = mockData;
    const props = {
      posts
    };

    wrapper = shallow(<FeaturedArticles {...props} />);
  });

  test('should render a snapshot of the userNavigation component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
