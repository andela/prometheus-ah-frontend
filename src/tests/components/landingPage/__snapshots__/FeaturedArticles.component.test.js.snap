// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage component should render a snapshot of the userNavigation component 1`] = `
<div
  className="col-12 col-md-7 col-lg-7"
>
  <h2
    className="section-title"
  >
    FEATURED POSTS
  </h2>
  <FeaturedArticle
    active={0}
    author={
      Object {
        "firstname": null,
        "lastname": null,
        "username": "faksam",
      }
    }
    category="Self"
    date="2018-10-20T12:21:33.812Z"
    description="all our problems"
    key="23"
    readTime="1 min read"
    slug="nobody-taught-you"
    title="nobody taught you"
  />
</div>
`;
